src/devices/bus/mc10/mc10cart.h

```diff
@@ -0,0 +1,133 @@
+
+#ifndef MAME_BUS_MC10_MC10CART_H
+#define MAME_BUS_MC10_MC10CART_H
+
+#pragma once
+
+#include "softlist_dev.h"
+
+/***************************************************************************
+    TYPE DEFINITIONS
+***************************************************************************/
+
+
+// ======================> mc10cart_slot_device
+class device_mc10cart_interface;
+
+class mc10cart_slot_device final : public device_t,
+								public device_single_card_slot_interface<device_mc10cart_interface>,
+								public device_image_interface
+{
+public:
+
+	// construction/destruction
+	template <typename T>
+	mc10cart_slot_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock, T &&opts, const char *dflt)
+		: mc10cart_slot_device(mconfig, tag, owner, clock)
+	{
+		option_reset();
+		opts(*this);
+		set_default_option(dflt);
+		set_fixed(false);
+	}
+
+	mc10cart_slot_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	// inline configuration
+	template <typename T> void set_memspace(T &&tag, int spacenum) { m_memspace.set_tag(std::forward<T>(tag), spacenum); }
+	auto nmi_callback() { return m_nmi_callback.bind(); }
+
+	// address map manipulations
+	address_space &memspace() const { return *m_memspace; };
+	void install_bank(offs_t start, offs_t end, uint8_t *data);
+	void install_rom(offs_t start, offs_t end, uint8_t *data);
+
+	// device-level overrides
+	virtual void device_start() override;
+
+	// image-level overrides
+	virtual image_init_result call_load() override;
+	virtual const software_list_loader &get_software_list_loader() const override { return rom_software_list_loader::instance(); }
+
+	virtual iodevice_t image_type() const noexcept override { return IO_CARTSLOT; }
+
+	virtual bool is_readable()  const noexcept override { return true; }
+	virtual bool is_writeable() const noexcept override { return false; }
+	virtual bool is_creatable() const noexcept override { return false; }
+	virtual bool must_be_loaded() const noexcept override { return false; }
+	virtual bool is_reset_on_load() const noexcept override { return true; }
+	virtual const char *image_interface() const noexcept override { return "mc10_cart"; }
+	virtual const char *file_extensions() const noexcept override { return "mcc,rom"; }
+
+	// manipulation of nmi line
+	void set_nmi_line(int state);
+	devcb_write_line m_nmi_callback;
+
+	// slot interface overrides
+	virtual std::string get_default_card_software(get_default_card_software_hook &hook) const override;
+
+private:
+
+	// cartridge
+	device_mc10cart_interface *m_cart;
+
+protected:
+	required_address_space m_memspace;
+};
+
+// device type definition
+DECLARE_DEVICE_TYPE(MC10CART_SLOT, mc10cart_slot_device)
+
+
+// ======================> device_mc10cart_host_interface
+
+// this is implemented by the MC-10 root device itself
+class device_mc10cart_host_interface
+{
+public:
+};
```

Comment:
What’s the purpose of this, since it has no members?  It looks like a simple bus anyway, no complex interactions with the host that would require some kind of interface rather than callback configuration.